test/e2e/farm_build_test.go

```diff
@@ -0,0 +1,1379 @@
+				expectBuildsOf{
+					build{arch: HOST_ARCH, expectedTobeBuiltOn: LOCAL_HOST, usingEmulation: false, withCleanup: false},
+				},
+			),
+			//
+			Entry("Multi Node Farm but with a single distinct architecture available on it",
+				withTestScenarioOf{farm: "multinodeFarm", params: "--local=true", image: standardTestImage, tag: GOOD_SHORT_TAG},
+				expectBuildsOf{
+					build{arch: HOST_ARCH, expectedTobeBuiltOn: LOCAL_HOST, usingEmulation: false, withCleanup: false},
+				},
+			),
+		)
+		/*##########################################################################################################*/
+		/*##########################################################################################################*/
+
+		DescribeTableSubtree("Failure Scenarios        :",
+
+			func(scenario withTestScenarioOf, expectedFailure expectFailureWith) {
+
+				var failedOperationTest = func(
+					podmanTestInteg *PodmanTestIntegration,
+					scenario withTestScenarioOf,
+					expectedFailure expectFailureWith) {
+
+					// Note: The test may be performed with either the a local
+					// or remote variant of podmanTest, so to make it explicit,
+					// we are passing that as a parameter rather than just
+					// accessing the global podmanTest (though we could)
+
+					// Substitute any macros in use
+					scenario = transformScenario(scenario)
+
+					// Run the build and capture the outputs for analysis
+					sessions := performFarmBuildAccordingToScenario(podmanTestInteg, scenario)
+
+					// We are expecting all these build to succeed, so first check
+					// that nothing is reporting any failures.
+					By("Check whether the farm build is reporting errors")
+					Expect(sessions.Bld).NotTo(indicateAnErrorFreeFarmBuild(scenario.tag), "indicateAnErrorFreeFarmBuild() unexpectedly succeeded")
+
+					// Now check whether the error message is as expected
+					Expect(sessions.Bld.ErrorToString()).Should(ContainSubstring(expectedFailure.message), "Unexpected Error message")
+				}
+
+				/*##########################################################################################################*/
+
+				It("using podman binary", func() {
+					failedOperationTest(podmanLocal.PodmanTestIntegration, scenario, expectedFailure)
+				})
+				/*##########################################################################################################*/
+
+				It("using podman-remote binary", func() {
+					failedOperationTest(podmanTest, scenario, expectedFailure)
+				})
+				/*##########################################################################################################*/
+			},
+			//
+			/* #############################################################################################################*/
+			/*                                                                                                              */
+			/*  Farm Build Test Scearios : scenarios expected to fail.                                                      */
```

Comment:
```suggestion
			/*  Farm Build Test Scenarios : scenarios expected to fail.                                                      */
```
and anywhere else I might have missed.

Reply:
Happy to tidy up as per the above.